src/Microsoft.ML.AutoML/API/BinaryClassificationExperiment.cs

```diff
@@ -346,6 +352,13 @@ public BinaryClassificationRunner(MLContext context, IDatasetManager datasetMana
             _rnd = settings.Seed.HasValue ? new Random(settings.Seed.Value) : new Random();
         }
 
+        public void Dispose()
+        {
+            _context.CancelExecution();
+            _context = null;
+            GC.SuppressFinalize(this);
```

Comment:
>SuppressFinalize [](http://example.com/codeflow?start=15&length=16)

I am not seeing this class has a finalizer. what exactly this is for?

Reply:
Oh this is automatically added by VS when I select "implement IDisposable with disposable pattern", I'll remove that as there's no finalizer for that class.